### PR TITLE
Remove unnecessary EM_CORE checks

### PR DESCRIPTION
**Pull request by weiwangncar**
Remove a few unnecessary EM_CORE checks

TYPE: no impact, code clean up

KEYWORDS: EM_CORE,  clean up, check_a_mundo

SOURCE: internal

DESCRIPTION OF CHANGES:
Problem:
Some EM_CORE checks in the routine are no longer needed.

Solution:
They are removed.

LIST OF MODIFIED FILES: 
M      share/module_check_a_mundo.F

TESTS CONDUCTED: 
1. Should have no impact. 
2. Are the Jenkins tests all passing?

RELEASE NOTE:

### CODE REVIEW

**Comment by weiwangncar** [comment]
The Jenkins tests have passed (including all fixes up to now in release-v4.4.1 branch):

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           58
    Number of Simulations  : 158           156        0
    Number of Comparisons  : 95           92        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

**Review by smileMchen — APPROVED** [review]

**Comment by weiwangncar** [comment]
> @weiwangncar Why is the modification to data.h included in this PR?

I was using this PR to test a few things, including the data.h change. This PR does not need to be included in 4.4.1 release. I will change the marking.

**Review by dudhia — COMMENTED** [review]
I have a pending comment. It is labeled as Do Not Merge, but I think it can be if this comment is answered. It is mostly just clean-up.

File: tools/data.h [hunk 1]
```diff
@@ -75,7 +75,7 @@ typedef struct node_struct {
   char pkg_4dscalars[NAMELEN_LONG] ;
 
 /* fields used by Comm (halo, period, xpose)  nodes */
-  char comm_define[2*8192] ;
+  char comm_define[4*8192] ;
```
[the remark this answers is not in the document]
Comment (weiwangncar) [reply tools/data.h:78]: @dudhia I created this PR to test the repository at the time. The line you pointed out is a fix we put in to correct moving nest compilation error which has been included in the repository (PR#1758 for 4.4.1). There are many clean-ups that can be done, so this is not complete at all.

[the remark this answers is not in the document]
Comment (mgduda) [reply tools/data.h:78]: I'd suggest reverting this change, as it has nothing to do with `EM_CORE` checks.

**Comment by weiwangncar** [comment]
May try again in the future.

**Pull request closed without merging**